docs/how_backtrader_executes_code.rst

```diff
@@ -0,0 +1,105 @@
+TLDR: How BackTrader Executes Code
+==================================
+
+Preonce, Once, Next, Init – Why Isn’t My Code Working?
+------------------------------------------------------
+
+Backtrader evolved from a backtesting only platform to a platform that performs real-time
+processing of data. It now has signal generation from strategies, optimization and trade
+execution. Because of this there are different ways for Backtrader to execute indicator code
+varying depending on whether or not the data is live or historical and preloaded.
+
+Coding for Backtesting:
+
+This is how Backtrader processes indicator code when backtesting preloaded data.
+
+By LINES:
+
+	Init() - processes bars as lines, all at once
+
+	Nothing else needed
+
+
+By BARS:
+
+	Init() - processes bars as lines, all at once
+
+	Next() - processes different individual bars based on above bars
+
+Or
+
+By BARS:
+
+	Init() - empty or just cursory setup for next
+
+	Next() = Run for each bar to caculate
+
+
+By BATCH:
+
+	Once - processes an array that you step through for	speed
+
+	It ends. It doesn’t run next(), doesn’t run init(), nothing. This is for speed
+
+
+That’s it – it doesn’t execute preonce, once, unless it’s running the once routine for this mode
+of execution. And when running once it doesn’t run init or next. And when running init and/or
+next it is not running once(). For simplicity this diagram excludes prenext and similar.
+
+So you have your file of stored data with timestamps and imported it with read_csv and
+adddata as various online examples show. You have your timestamps in UTC time and you
+made sure that they are going from earliest to latest so that backtrader doesn’t try to start
+from the future. Great! You code your routine in next() so that it executes at each bar.
+
+...and it runs!
+
+Fantastic! But it’s slow. So you add some preconfiguration in once() to help it along a bit.
+And it explodes. You might think “is this platform broken?” Why isn’t once running when the
+bars have reached the minimum period and then next running for each bar? Why? Because
+that’s not what backtrader is designed to do. Once is only for backtesting. Isolated on its own
+with its brothers preonce, oncestart, etc.
+
+So, fine, you move all your code to the once routine but... wait... you spend hours diagnosing
```

Comment:
I'm kind of lost here.  It is probably worth working with some code samples throughout the doc to illustrate the phases of development.

Reply:
Code samples are covered in the docs with an SMA as I recall. Perhaps this could pasted interspersed with the existing "Operating the Platform" page under "Line Iterators"